library/std/src/sys/pal/unix/stack_overflow.rs

```diff
@@ -186,34 +206,45 @@ mod imp {
         libc::stack_t { ss_sp: stackp, ss_flags: 0, ss_size: sigstack_size }
     }
 
+    /// # Safety
+    /// Mutates the alternate signal stack
+    #[forbid(unsafe_op_in_unsafe_fn)]
     pub unsafe fn make_handler(main_thread: bool) -> Handler {
         if !NEED_ALTSTACK.load(Ordering::Relaxed) {
             return Handler::null();
         }
 
         if !main_thread {
             // Always write to GUARD to ensure the TLS variable is allocated.
-            let guard = current_guard().unwrap_or(0..0);
+            let guard = unsafe { current_guard() }.unwrap_or(0..0);
```

Comment:
it's not entirely clear to me why `current_guard` is unsafe, and this doesn't seem like a safety comment even if you squint, thus no `// SAFETY` here.

Reply:
I agree. Could you add a FIXME to `current_guard` stating that the `unsafe` should be removed?